odxtools/admindata.py

```diff
@@ -51,15 +54,19 @@ def _build_odxlinks(self) -> Dict[OdxLinkId, Any]:
 
         return result
 
-    def _resolve_references(self, odxlinks: OdxLinkDatabase):
+    def _resolve_odxlinks(self, odxlinks: OdxLinkDatabase):
         self._company_data = odxlinks.resolve(self.company_data_ref, CompanyData)
 
         self._team_member: Optional[TeamMember] = None
         if self.team_member_ref is not None:
             self._team_member = odxlinks.resolve(self.team_member_ref, TeamMember)
 
         for sdg in self.sdgs:
-            sdg._resolve_references(odxlinks)
+            sdg._resolve_odxlinks(odxlinks)
+
+    def _resolve_snrefs(self, diag_layer: "DiagLayer"):
```

Comment:
Why not just one function that takes a DiagLayer and OdxLinkDatabase as arguments?

Reply:
That would be correct if any ODX Link resolution would cause a change in SNREF resolution, but I don't see how that could happen.

Reply:
for example by not inheriting a DIAG-COMM that is only referenced by the base layer...

Reply:
ODX Link uses doc REF to decided what to actually use (by default they always use same document unless there is a DOC-REF), I don't think they are meant to be affected by diag layer inheritance.

Reply:
yes, but if you have e.g., an ECU-INSTANCE layer `ECU` which does not inherit the diag-comm `FOO` from its parent BASE-VARIANT `BASE` and `BASE` references `FOO` from somewhere else instead of defining it, you must have resolved `FOO` in `BASE` before being able to determine which diag comms `ECU` ought to inherit... (if you ask me, the spec is way too complicated and I doubt that it is implemented 100% correctly by *any* software package...)